Lib/sqlite3/dump.py

```diff
@@ -26,6 +26,7 @@ def _iterdump(connection):
 
     writeable_schema = False
     cu = connection.cursor()
+    yield('PRAGMA foreign_keys=OFF;')
```

Comment:
This is consistent with the sqlite CLI .dump but seems to contradict the documentation [here](https://www.sqlite.org/foreignkeys.html) (section 2). Perhaps we should query `foreign_keys` for future proofing?

Excerpt from section 2:
Foreign key constraints are disabled by default (for backwards compatibility), so must be enabled separately for each [database connection](https://www.sqlite.org/c3ref/sqlite3.html). (Note, however, that future releases of SQLite might change so that foreign key constraints enabled by default. Careful developers will not make any assumptions about whether or not foreign keys are enabled by default but will instead enable or disable them as necessary.)

Reply:
I don't we can make assumptions about a connection on which the dump will be imported. We can check if there is any foreign key violation on the source database before adding this, e.g.
```python
violations = cursor.execute("PRAGMA foreign_key_check").fetchall()
if len(violations) > 0:
    yield('PRAGMA foreign_keys=OFF;')
```

Reply:
@erlend-aasland  Do you need help this this patch?

Reply:
Updated in #113957.